src/Analysis/Ast/Impl/Specializations/Abc/AbcModule.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections.Generic;
+using Microsoft.Python.Analysis.Modules;
+using Microsoft.Python.Analysis.Types;
+using Microsoft.Python.Core;
+
+namespace Microsoft.Python.Analysis.Specializations.Abc {
+    internal sealed class AbcModule : SpecializedModule {
+        private readonly Dictionary<string, IMember> _members = new Dictionary<string, IMember>();
+
+        private AbcModule(string modulePath, IServiceContainer services) : base("abc", modulePath, services) { }
+
+        public static IPythonModule Create(IServiceContainer services) {
+            var interpreter = services.GetService<IPythonInterpreter>();
+            var module = interpreter.ModuleResolution
+                .SpecializeModule("abc", modulePath => new AbcModule(modulePath, services)) as AbcModule;
+            module?.SpecializeMembers();
+            return module;
+        }
+
+        #region IMemberContainer
+        public override IMember GetMember(string name) => _members.TryGetValue(name, out var m) ? m : null;
+        public override IEnumerable<string> GetMemberNames() => _members.Keys;
+        #endregion
+
+        private void SpecializeMembers() {
+            // ABC
+            var cls = PythonClassType.Specialize("ABC", this, GetMemberDocumentation("ABC"), isAbstract: true);
```

Comment:
If we specialize module, we no longer look into the original one. Are there any members which we may lose as we only specialize one?

Reply:
We lose ABCMeta, which I plan to handle once I figure out how to handle metaclasses. But I believe that is it. Relevant link is: https://github.com/python/cpython/blob/master/Lib/abc.py

Reply:
So I guess this PR should be in `WIP` unless we are sure users won't complain that `ABCMeta` is no longer in completions?

Reply:
Yup, I'll do that.